exchange/bitswap/workers.go

```diff
@@ -215,35 +276,22 @@ func (bs *Bitswap) providerQueryManager(ctx context.Context) {
 			default:
 			}
 
-			activeLk.Lock()
+			ksetLk.Lock()
 			if kset.Has(e.Cid) {
```

Comment:
might want to use `Visit` here: https://github.com/ipfs/go-cid/blob/master/set.go#L47